model/project.go

```diff
@@ -1450,20 +1472,56 @@ func (p *Project) BuildProjectTVPairs(patchDoc *patch.Patch, alias string) {
 // mapping of the build variant to the tasks that will run on that build
 // variant. If includeDeps is set, it will also resolve task dependencies.
 func (p *Project) ResolvePatchVTs(patchDoc *patch.Patch, requester, alias string, includeDeps bool) (resolvedBVs []string, resolvedTasks []string, vts []patch.VariantTasks) {
-	bvs := patchDoc.BuildVariants
-	tasks := patchDoc.Tasks
-	if len(bvs) == 1 && bvs[0] == "all" {
-		bvs = []string{}
-		for _, bv := range p.BuildVariants {
+	var bvs, tasks []string
+	for _, bv := range patchDoc.BuildVariants {
+		var bvSearch []BuildVariant
+		switch bv {
+		case "all":
+			bvSearch = p.BuildVariants
+		case "":
+			continue
+		default:
+			bvRegex, err := regexp.Compile(bv)
+			if err != nil {
+				grip.Error(message.WrapError(err, message.Fields{
+					"message": "compiling buildvariant regex",
+					"regex":   bv,
+					"project": p.Identifier,
+				}))
+				continue
+			}
+			bvSearch = p.FindMatchingBuildVariants(bvRegex)
+		}
+		for _, bv := range bvSearch {
 			if bv.Disabled {
 				continue
 			}
 			bvs = append(bvs, bv.Name)
 		}
 	}
-	if len(tasks) == 1 && tasks[0] == "all" {
-		tasks = []string{}
-		for _, t := range p.Tasks {
+	for _, t := range patchDoc.Tasks {
+		var tSearch []ProjectTask
+		switch t {
+		case "all":
+			tSearch = p.Tasks
+		case "":
+			continue
+		default:
+			tRegex, err := regexp.Compile(t)
+			if err != nil {
+				grip.Error(message.WrapError(err, message.Fields{
+					"message": "compiling task regex",
+					"regex":   t,
+					"project": p.Identifier,
+				}))
+				continue
+			}
+			tSearch = p.FindMatchingProjectTasks(tRegex)
+			if len(tSearch) == 0 {
```

Comment:
This is kind of weird. Why does this have to check for no matching project tasks but the build variant logic above this doesn't?

Reply:
the new fix doesnt not need this logic anymore